dashboards/argo-workflows/argo-workflows-prometheus.json

```diff
@@ -17,115 +17,125 @@
-        "xPos": 0,
-        "yPos": 0
+          }
+        }
       },
       {
-        "height": 2,
+        "xPos": 40,
+        "width": 8,
+        "height": 8,
         "widget": {
+          "title": "Errors",
           "scorecard": {
             "blankView": {},
+            "thresholds": [],
             "timeSeriesQuery": {
-              "prometheusQuery": "argo_workflows_count{status=\"Error\",${Cluster},${Location},${Namespace}}"
+              "prometheusQuery": "argo_workflows_count{status=\"Error\",${Cluster},${Location},${Namespace}}",
+              "unitOverride": ""
             }
-          },
-          "title": "Errors"
-        },
-        "width": 2,
-        "xPos": 10,
-        "yPos": 0
+          }
+        }
       },
       {
-        "height": 2,
+        "xPos": 32,
+        "width": 8,
+        "height": 8,
         "widget": {
+          "title": "Failed",
           "scorecard": {
             "blankView": {},
+            "thresholds": [],
             "timeSeriesQuery": {
-              "prometheusQuery": "argo_workflows_count{status=\"Failed\",${Cluster},${Location},${Namespace}}"
+              "prometheusQuery": "argo_workflows_count{status=\"Failed\",${Cluster},${Location},${Namespace}}",
+              "unitOverride": ""
             }
-          },
-          "title": "Failed"
-        },
-        "width": 2,
-        "xPos": 8,
-        "yPos": 0
+          }
+        }
       },
       {
-        "height": 2,
+        "xPos": 16,
+        "width": 8,
+        "height": 8,
         "widget": {
+          "title": "Succeeded",
           "scorecard": {
             "blankView": {},
+            "thresholds": [],
             "timeSeriesQuery": {
-              "prometheusQuery": "argo_workflows_count{status=\"Succeeded\",${Cluster},${Location},${Namespace}}"
+              "prometheusQuery": "argo_workflows_count{status=\"Succeeded\",${Cluster},${Location},${Namespace}}",
+              "unitOverride": ""
             }
-          },
-          "title": "Succeeded"
-        },
-        "width": 2,
-        "xPos": 6,
-        "yPos": 0
+          }
+        }
       },
       {
-        "height": 4,
+        "yPos": 8,
+        "width": 24,
+        "height": 10,
         "widget": {
-          "title": "Operation Duration (seconds)",
+          "title": "Operation Duration Seconds",
           "xyChart": {
             "chartOptions": {
               "mode": "COLOR"
             },
             "dataSets": [
               {
+                "breakdowns": [],
+                "dimensions": [],
+                "measures": [],
                 "plotType": "LINE",
                 "targetAxis": "Y1",
                 "timeSeriesQuery": {
-                  "prometheusQuery": "argo_workflows_operation_duration_seconds_sum{${Cluster},${Location},${Namespace}}"
+                  "prometheusQuery": "argo_workflows_operation_duration_seconds_bucket{${Cluster},${Location},${Namespace}}",
```

Comment:
_bucket metrics in promQL generally need some kind of aggregation applied to display well. What if we call this chart "Median Operation Duration" and use:

`histogram_quantile(0.50,rate(argo_workflows_operation_duration_seconds_bucket{${Cluster},${Location},${Namespace}}[${__interval}]))`

With the query as-is, we can see in the updated screenshot that the legend has a bunch of numbers which is kind of confusing to read - it's showing a separate line for how many operations happen in each duration bucket, with buckets like <.1 seconds, .1-.25 seconds, and finally up to +Infinity seconds.

I've found it helpful to build promQL for these bucket metrics by first building the query in the basic query builder mode, then converting it automatically to promQL.
![image](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/assets/509446/caaa3408-b455-4a57-86ee-40a54ea368c0)

Reply:
@johnbryan Thanks for the explanation here, it was helpful and I appreciate it. I've updated both the queries.